Xamarin.Forms.Core/MultiBinding.cs

```diff
@@ -70,238 +66,147 @@ internal override BindingBase Clone()
+				return;
 
-			if (IsApplied && fromBindingContextChanged)
-			{
-				bool childContextChanged = false;
-				foreach (var proxy in SourceProxies)
-				{
-					if (!object.ReferenceEquals(proxy.BindingContext, context))
-					{
-						childContextChanged = true;
-						proxy.SetValueSilent(Element.BindingContextProperty, context);
-					}
-				}
-				if ( childContextChanged )
-					ApplyBindingProxyValues(null, true);
+			if (!fromTarget && this.GetRealizedMode(_targetProperty) == BindingMode.OneWayToSource)
 				return;
-			}
-		
-			_targetProperty = targetProperty;
 			
-			if (_expression == null)
-				_expression = new BindingExpression(this, nameof(MultiBindingProxy.Value));
-
-			CreateBindingProxies(bindObj, context);
-
-			_isApplying = true;			
-			try
+			if (!fromTarget)
 			{
-				ApplyBindingProxyValues(TargetProxy, reapplyExpression: false, firstApplication: true);
-				_expression.Apply(TargetProxy, bindObj, targetProperty);
-				if (this.GetRealizedMode(_targetProperty) == BindingMode.OneWayToSource &&
-					 this.FallbackValue != null)
-					bindObj.SetValue(targetProperty, this.FallbackValue);
+				var value = GetSourceValue(GetValueArray(), _targetProperty.ReturnType);
+				if (value != Binding.DoNothing) {
+					_applying = true;
+					_targetObject.SetValueCore(_targetProperty, value, SetValueFlags.ClearDynamicResource, BindableObject.SetValuePrivateFlags.Default | BindableObject.SetValuePrivateFlags.Converted);
+					_applying = false;
+				}
 			}
-			finally
+			else
 			{
-				_isApplying = false;
+				try {
+					_applying = true;
+
+					//https://docs.microsoft.com/en-us/dotnet/api/system.windows.data.imultivalueconverter.convertback?view=netframework-4.8#remarks
+					if (!(GetTargetValue(_targetObject.GetValue(_targetProperty), null) is object[] values)) //converter failed
+						return;
+					for (var i = 0; i < Math.Min(_bpProxies.Length, values.Length); i++)
+					{
+						if (ReferenceEquals(values[i], Binding.DoNothing) || ReferenceEquals(values[i], BindableProperty.UnsetValue))
+							continue;
+						_proxyObject.SetValueCore(_bpProxies[i], values[i], SetValueFlags.None);
+					}
+				}
+				finally {
+					_applying = false;
+				}
+
 			}
 		}
 
-		internal override void Unapply(bool fromBindingContextChanged = false)
+		internal override void Apply(object context, BindableObject targetObject, BindableProperty targetProperty, bool fromBindingContextChanged = false)
 		{
-			if (fromBindingContextChanged && IsApplied)
-				return;
-
-			base.Unapply(fromBindingContextChanged: fromBindingContextChanged);
+			if (_bindings == null)
+				throw new InvalidOperationException("Bindings is null");
+
+			base.Apply(context, targetObject, targetProperty, fromBindingContextChanged);
+
+			if (!ReferenceEquals(_targetObject, targetObject)) {
+				_targetObject = targetObject;
+				_proxyObject = new ProxyElement() { Parent = targetObject as Element };
```

Comment:
ProxyElement is our lightweight proxy. it's an element so it has a parent (for RelativeSource Bindings)